lib/compress/zstd_compress.c

```diff
@@ -2928,10 +3028,57 @@ static size_t ZSTD_buildSeqStore(ZSTD_CCtx* zc, const void* src, size_t srcSize)
                                        zc->appliedParams.useRowMatchFinder,
                                        src, srcSize);
             assert(ldmSeqStore.pos == ldmSeqStore.size);
-        } else {   /* not long range mode */
+        } else if (zc->appliedParams.useExternalMatchFinder) {
+            assert(
+                zc->externalMatchCtx.seqBufferCapacity >= ZSTD_sequenceBound(srcSize)
+            );
+            assert(zc->externalMatchCtx.mFinder != NULL);
+
+            {   size_t const nbExternalSeqs = (zc->externalMatchCtx.mFinder)(
+                    zc->externalMatchCtx.mState,
+                    zc->externalMatchCtx.seqBuffer,
```

Comment:
Could you memset all values of array zc->externalMatchCtx.seqBuffer to 0 before passing it to matchfinder?

Reply:
This is difficult because normally the matchfinder will only use a tiny fraction of the seqBuffer. So there could be a large performance cost if we memset the entire buffer.

Why do you need to memset to zero?

Reply:
`seqBuffer` should be write only, meaning its initial state should not matter, hence not be relied upon.
If you believe you need this condition, could you explain why ?

Reply:
No need to set it to zero now. I met a bug in my qatmatchfinder implementation, so I set it to zero. Now I fixed my bug.